_posts/en/newsletters/2022-07-13-newsletter.md

```diff
@@ -0,0 +1,173 @@
+    well.
+
+    This week, Tim Ruffing [posted][ruffing xonly] to the Bitcoin-Dev
+    mailing list a potential workaround that only requires a slight bit
+    of additional computation by signers---an amount that even a
+    resource-constrained hardware signing device can probably perform
+    without making its user wait too long.
+
+- **Allowing deliberately slow LN payment forwarding:** in a reply to a
+  thread about recursive/nested MuSig2 (see [Newsletter #204][news204
+  rmusig]) and the latency that nodes using it would add when routing
+  payments, developer Peter Todd [asked][todd delay] on the
+  Lightning-Dev mailing list whether it would "be worthwhile to allow
+  people to opt-in to their payments happening more slowly for privacy?"
+  For example, if Alice and Bob each sent forward-slowly payments around
+  the same time through Carol's forwarding node to Dan's forwarding
+  node, Carol would be able to forward both payments together, reducing
+  the amount of privacy-leaking information about the participants that
+  third parties could discover through [balance probing][FIXME add
+  topic], network activity surveillance, or other techniques.  Developer
+  Matt Corallo [agreed][corallo delay] it was an interesting idea.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+FIXME:glozow
+
+{% include functions/details-list.md
+
+  q0="FIXME"
+  a0="FIXME"
+  a0link="https://bitcoincore.reviews/24148#l-41FIXME"
+
+%}
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+FIXME
+
+- BTCPay Server 1.6.0 FIXME
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #25353][] introduces the `mempoolfullrbf` configuration
+  option which may be used to generally permit transaction replacement.
+  Currently, Bitcoin Core will only allow replacement of transactions in
+  its mempool that have signaled replaceability as defined per
+  [BIP125][]. This behavior is referred to as "opt-in full RBF" which
+  continues to be the default behavior of nodes after this change. The
+  new `mempoolfullrbf` option permits a node operator to change their
+  node's replacement behavior to "general full RBF" which permits
```

Comment:
Question, would you be able to point to where {opt-in, general} {full, not full} RBF terms are defined?

Reply:
For opt-in RBF maybe:

- https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2015-November/011783.html
- https://bitcoincore.org/en/faq/optin_rbf/
- https://bitcoin.stackexchange.com/q/38320/5406

I'm not sure whether anyone else previously used "general full RBF". The alternative proposal was called "first-seen-safe RBF", which required that all the outputs of the replaced transactions had to be present in the replacement transaction with the same or a higher amount.

Reply:
I think this summary might benefit by linking to when announce this was a possible upcoming change in https://bitcoinops.org/en/newsletters/2022/06/22/#full-replace-by-fee .  I think that could reduce the likeliness that people would get upset by your description---we can just point out that we warned people that the idea was on the table.  (We also warned them last time this came up too: https://bitcoinops.org/en/newsletters/2021/06/23/#allowing-transaction-replacement-by-default )

I think a link to our RBF topic page might also be helpful: https://bitcoinops.org/en/topics/replace-by-fee/

---

Regarding terminology, if anyone is feeling ambitious, maybe open a PR to our RBF topic page where you define the terms you think are relevant.  If you make it a list with bold terms (like in https://bitcoinops.org/en/topics/op_checksigfromstack/), you'll be able to link to the specific terms.

I don't care what term is used here but historically Peter Todd, who chose the terms, has used "opt-in full-RBF" for (what became) BIP125 and "full-RBF" for signaling-not-required replacements.  See, e.g., https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2017-January/013471.html

In Optech, I've preferred to use "opt-in RBF" for BIP125 and "full RBF" for non-signaled, e.g. https://bitcoinops.org/en/newsletters/2022/06/22/#full-replace-by-fee

Reply:
> Peter Todd, who chose the terms, has used "opt-in full-RBF" for (what became) BIP125 and "full-RBF" for signaling-not-required replacements. See, e.g., https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2017-January/013471.html

> In Optech, I've preferred to use "opt-in RBF" for BIP125 and "full RBF" for non-signaled, e.g. https://bitcoinops.org/en/newsletters/2022/06/22/#full-replace-by-fee

This is also how I use them, i.e. opt-in means you apply signaling rules, full means you don't. I don't think we should  use "general full rbf" if it's a new term.

Reply:
Thank you for being open-minded! :) I learned something new about RBF history from this discussion. The summary now looks :ok_hand: to me.